js/components/report/iframe-answer.tsx

```diff
@@ -159,44 +153,63 @@ class IframeAnswer extends PureComponent<IProps, IState> {
     }
 
     if (reportItemAnswer) {
-      switch (reportItemAnswer.type) {
-        case "html":
-          reportItemHTML = `
-            <!DOCTYPE html>
-            <html>
-            <head>
-              <style>
-                body {
-                  font-family: lato, arial, helvetica, sans-serif;
-                }
-                .tall {
-                  display: ${displayTall ? "flex" : "none"};
-                  flex-direction: column;
-                }
-                .wide {
-                  display: ${displayTall ? "none" : "flex"};
-                  flex-direction: row;
-                }
-              </style>
-            </head>
-            <body>
-              ${reportItemAnswer.html}
-            </body>
-            </html>
-          `;
-          break;
+      // handle interactives using the older api format and create a single html item
+      if (reportItemAnswer.type === "html") {
+        reportItemAnswerItems.push({
+          type: "html",
+          html: `
+          <!DOCTYPE html>
+          <html>
+          <head>
+            <style>
+              body {
+                font-family: lato, arial, helvetica, sans-serif;
+              }
+              .tall {
+                display: ${displayTall ? "flex" : "none"};
+                flex-direction: column;
+              }
+              .wide {
+                display: ${displayTall ? "none" : "flex"};
+                flex-direction: row;
+              }
+            </style>
+          </head>
+          <body>
+            ${reportItemAnswer.html}
+          </body>
+          </html>
+        `});
+      } else {
+        // use the new api format which returns a list of items
+        reportItemAnswerItems = reportItemAnswer.items || [];
       }
     }
 
+    // if there are no report answer items, show the answer text or the links to view the interactive
+    let maybeAnswerTextOrLinks: JSX.Element | false = false;
+    if (reportItemAnswerItems.length === 0) {
+      maybeAnswerTextOrLinks = answerText
+        ? <div data-cy="answerText">{ renderHTML(answerText) }</div>
+        : !alwaysOpen && this.renderLink(); /* This assumes only scaffolded questions and fill in the blank questions have answerTexts */
```

Comment:
I see. Would a reasonable approach to the answerText problem be to change linies 142 - 143 to something like the below?

`const questionType = answer.get("questionType");
  const answerText = answer.get("answerText")
                         ? answer.get("answerText")
                         : questionType === "open_response"
                           ? answer.get("answer")
                           : undefined;`

Reply:
Instead of changing iframe-answer, I'd try using updating the AnswerComponent map.  It is currently just looking at the answer type and then returning a component. Instead it could look at both the type and perhaps the presence of the some field that all interactives have.  So then when the answer is from a built in open response it would return a OpenResponseAnswer and when the answer is from a interactive open response it would return the IframeAnswer.

Probably this means it'd be best to extract this into a function like: `getAnswerComponent(answer)`

Reply:
@scytacki Thanks. I've essentially made those changes in the latest commit. One issue: there isn't anything in the Answer component's props that would identify the question as being LARA-native question or a managed interactive. For now, I used a check for answerText on open response question answers. That seems to work OK in the browser, but is there a reason that's not ideal? Should I instead find a way to pass down to the Answer component information about the question being LARA-native or a managed interactive?

Reply:
It seems using answerText should be fine.